src/github.com/getlantern/rotator/size_rotator.go

```diff
@@ -40,24 +40,25 @@ func (r *SizeRotator) Write(bytes []byte) (n int, err error) {
 
 	// Do rotate when size exceeded
 	if r.totalSize+int64(len(bytes)) > r.RotationSize {
-		// Get available file name to be rotated
-		for i := 1; i <= r.MaxRotation; i++ {
-			renamedPath := r.path + "." + strconv.Itoa(i)
-			stat, _ := os.Lstat(renamedPath)
-			if stat == nil {
-				err := os.Rename(r.path, renamedPath)
-				if err != nil {
-					return 0, err
-				}
-				if r.file != nil {
-					// reset file reference
-					r.file.Close()
-					r.file = nil
-				}
-				break
+		r.file.Close()
```

Comment:
What if `r.file == nil`?  Since you've checked for that some lines up, I guess that's a possible condition?